### PR TITLE
Convert all value of tag to string.

### LINKED ISSUE #8: Reporter flush failed when using with Jaeger backend
When I use Kong Zipkin plugin with Jaeger backend, I got an error with the response body:

`Unable to process request body: json: cannot unmarshal number into Go struct field Span.tags of type string`

Kong version: 0.14.0rc1 on docker.

Request body of Reporter:
```json
[
    {
        "traceId": "9aa46f78b4afbd37e6845ba62313d51d",
        "localEndpoint": null,
        "id": "d4097ab8438f67b0",
        "name": "kong.rewrite",
        "parentId": "ec270a5446cd8a68",
        "remoteEndpoint": null,
        "duration": 0,
        "timestamp": 1.530259357671e+15
    },
    {
        "traceId": "9aa46f78b4afbd37e6845ba62313d51d",
        "localEndpoint": null,
        "id": "8a63117b77519823",
        "name": "kong.header_filter",
        "parentId": "842b5d62df16ba9d",
        "remoteEndpoint": null,
        "duration": 0,
        "timestamp": 1.530259357673e+15
    },
    {
        "traceId": "9aa46f78b4afbd37e6845ba62313d51d",
        "localEndpoint": null,
        "id": "6188f6111cb62a67",
        "name": "kong.access",
        "parentId": "842b5d62df16ba9d",
        "remoteEndpoint": null,
        "duration": 999,
        "timestamp": 1.530259357671e+15
    },
    {
        "traceId": "9aa46f78b4afbd37e6845ba62313d51d",
        "localEndpoint": null,
        "id": "16517979531c5062",
        "tags": {
            "kong.balancer.try": 1,
            "peer.port": 9500,
            "peer.ipv4": "10.148.7.194"
        },
        "name": "kong.balancer",
        "parentId": "842b5d62df16ba9d",
        "remoteEndpoint": {
            "ipv4": "10.148.7.194",
            "port": 9500
        },
        "duration": 0,
        "timestamp": 1.530259357672e+15
    },
    {
        "traceId": "9aa46f78b4afbd37e6845ba62313d51d",
        "localEndpoint": null,
        "id": "cc23407a6eb33e4f",
        "name": "kong.body_filter",
        "parentId": "842b5d62df16ba9d",
        "remoteEndpoint": null,
        "duration": 0,
        "timestamp": 1.530259357673e+15
    },
    {
        "traceId": "9aa46f78b4afbd37e6845ba62313d51d",
        "localEndpoint": null,
        "id": "842b5d62df16ba9d",
        "tags": {
            "peer.service": "test",
            "peer.port": 9500,
            "peer.ipv4": "10.148.7.194",
            "kong.service": "b1a71321-3b6d-46ef-b376-11e7275c3f36",
            "kong.route": "107bb250-a65f-4915-8731-1d7b860bdba7",
            "peer.hostname": "10.148.7.194",
            "span.kind": "client"
        },
        "name": "kong.proxy",
        "parentId": "ec270a5446cd8a68",
        "kind": "CLIENT",
        "remoteEndpoint": {
            "ipv4": "10.148.7.194",
            "port": 9500
        },
        "duration": 2000,
        "timestamp": 1.530259357671e+15
    },
    {
        "traceId": "9aa46f78b4afbd37e6845ba62313d51d",
        "localEndpoint": null,
        "id": "ec270a5446cd8a68",
        "tags": {
            "http.status_code": 400,
            "span.kind": "server",
            "peer.ipv4": "172.17.0.1",
            "http.method": "GET",
            "peer.port": 47580,
            "http.url": "http:\/\/localhost:8000\/test",
            "component": "kong"
        },
        "name": "kong.request",
        "kind": "SERVER",
        "remoteEndpoint": {
            "ipv4": "172.17.0.1",
            "port": 47580
        },
        "duration": 2000,
        "timestamp": 1.530259357671e+15
    }
]
```

### PR DESCRIPTION
**Pull request by bnlong4417**
This fixes #8

### CODE REVIEW

File: kong/plugins/zipkin/opentracing.lua [hunk 1]
```diff
@@ -171,18 +171,18 @@ function OpenTracingHandler:log(conf)
 			local try = balancer_tries[i]
 			local span = proxy_span:start_child_span("kong.balancer", try.balancer_start / 1000)
 			span:set_tag(ip_tag(try.ip), try.ip)
-			span:set_tag("peer.port", try.port)
-			span:set_tag("kong.balancer.try", i)
+			span:set_tag("peer.port", tostring(try.port))
+			span:set_tag("kong.balancer.try", tostring(i))
 			if i < balancer_address.try_count then
-				span:set_tag("error", true)
+				span:set_tag("error", "true")
```
Comment (james-callahan) [inline kong/plugins/zipkin/opentracing.lua]: According to the [opentracing spec](https://github.com/opentracing/specification/blob/master/semantic_conventions.md#span-tags-table) this is meant to be a boolean.

File: kong/plugins/zipkin/opentracing.lua [hunk 2]
```diff
@@ -171,18 +171,18 @@ function OpenTracingHandler:log(conf)
...
 				span:set_tag("kong.balancer.state", try.state)
-				span:set_tag("kong.balancer.code", try.code)
+				span:set_tag("kong.balancer.code", tostring(try.code))
```
Comment (james-callahan) [inline kong/plugins/zipkin/opentracing.lua]: The formatting of `tostring` is undefined; could you use `string.format("%d", try.port)` instead? (goes for elsewhere too)

File: kong/plugins/zipkin/opentracing.lua [hunk 3]
```diff
@@ -58,7 +58,7 @@ function OpenTracingHandler:initialise_request(conf, ctx)
 			["http.url"] = ngx.var.scheme .. "://" .. ngx.var.host .. ":" .. ngx.var.server_port .. ngx.var.request_uri;
 			[ip_tag(ngx.var.remote_addr)] = ngx.var.remote_addr;
 			["peer.ipv6"] = nil;
-			["peer.port"] = tonumber(ngx.var.remote_port, 10);
+			["peer.port"] = ngx.var.remote_port;
```
Comment (james-callahan) [inline kong/plugins/zipkin/opentracing.lua]: According to the [opentracing spec](https://github.com/opentracing/specification/blob/master/semantic_conventions.md#span-tags-table) this should be an integer.

**Comment by bnlong4417** [comment]
Yeah, we should fix [here](https://github.com/Kong/kong-plugin-zipkin/blob/master/kong/plugins/zipkin/reporter.lua#L49)

File: kong/plugins/zipkin/opentracing.lua [hunk 4]
```diff
@@ -172,11 +172,11 @@ function OpenTracingHandler:log(conf)
 			local span = proxy_span:start_child_span("kong.balancer", try.balancer_start / 1000)
 			span:set_tag(ip_tag(try.ip), try.ip)
 			span:set_tag("peer.port", try.port)
-			span:set_tag("kong.balancer.try", i)
+			span:set_tag("kong.balancer.try", string.format("%d", i))
 			if i < balancer_address.try_count then
 				span:set_tag("error", true)
 				span:set_tag("kong.balancer.state", try.state)
-				span:set_tag("kong.balancer.code", try.code)
+				span:set_tag("kong.balancer.code", string.format("%d", try.code))
```
Comment (james-callahan) [inline kong/plugins/zipkin/opentracing.lua]: I don't see any reason to set these tag values as strings (it seems more natural to keep them as numbers)

File: kong/plugins/zipkin/opentracing.lua [hunk 5]
```diff
@@ -194,7 +194,7 @@ function OpenTracingHandler:log(conf)
 		opentracing.body_filter_span:finish(ctx.KONG_BODY_FILTER_ENDED_AT / 1000)
 	end
 
-	request_span:set_tag("http.status_code", ngx.status)
+	request_span:set_tag("http.status_code", string.format("%d", ngx.status))
```
Comment (james-callahan) [inline kong/plugins/zipkin/opentracing.lua]: `http.status_code` is also meant to be an integer (See https://github.com/opentracing/specification/blob/master/semantic_conventions.md)

File: kong/plugins/zipkin/reporter.lua [hunk 6]
```diff
@@ -25,6 +25,26 @@ local span_kind_map = {
 	producer = "PRODUCER";
 	consumer = "CONSUMER";
 }
+
+-- Utility function to convert opentracing tags to zipkin tags
+-- values of zipkin tags should be strings, see [zipkin-api](https://zipkin.io/zipkin-api).
+function to_zipkin_tags(opentracing_tags)
+	zipkin_tags = {}
```
Comment (james-callahan) [inline kong/plugins/zipkin/reporter.lua]: Needs to be `local`

File: kong/plugins/zipkin/reporter.lua [hunk 7]
```diff
@@ -25,6 +25,26 @@ local span_kind_map = {
...
+	if not opentracing_tags then
```
Comment (james-callahan) [inline kong/plugins/zipkin/reporter.lua]: Needs to be `local`

File: kong/plugins/zipkin/reporter.lua [hunk 8]
```diff
@@ -25,6 +25,26 @@ local span_kind_map = {
...
+		return zipkin_tags
+	end
+
+	for k, v in pairs(opentracing_tags) do
```
Comment (james-callahan) [inline kong/plugins/zipkin/reporter.lua]: Use [`span:each_tag()`](https://github.com/Kong/opentracing-lua/blob/ee1cdc12b32097f6b9b4379e3ce207ecfaab2988/opentracing/span.lua#L89) here (anything else is relying on undefined internals)

File: kong/plugins/zipkin/reporter.lua [hunk 9]
```diff
@@ -25,6 +25,26 @@ local span_kind_map = {
...
+		if type(v) ~= "string" then
```
Comment (james-callahan) [inline kong/plugins/zipkin/reporter.lua]: Might as well use `tostring` unconditionally: it will return a string unchanged.

**Comment by adriancole** [comment]
so yeah in zipkin tags are strings. I think we might tolerantly read some values into strings on the sever side, but yeah string is what they should be. https://zipkin.io/zipkin-api/#/default/post_spans

File: kong/plugins/zipkin/reporter.lua [hunk 10]
```diff
@@ -25,6 +25,17 @@ local span_kind_map = {
 	producer = "PRODUCER";
 	consumer = "CONSUMER";
 }
+
+-- Utility function to get zipkin tags from opentracing span.
```
Comment (james-callahan) [inline kong/plugins/zipkin/reporter.lua]: I don't it's worth having this as it's own function (as it only has a single call-site).
Could you inline it into the `:report` method?

Comment (bnlong4417) [reply kong/plugins/zipkin/reporter.lua]: It's Ok 👍

**Pull request closed without merging**